lib/read_only/dl_esm_inf/read_only.f90

```diff
@@ -76,11 +72,7 @@ module read_only_verify_psy_data_mod
         !! to change the state from checksum computation to verification.
         !! And again the base class provides additional, unused methods
         !! like ProvideArray2dInt, which are not needed
-        generic, public :: ProvideVariable => ProvideScalarInt,    &
-                                              ProvideScalarReal,   &
-                                              ProvideScalarDouble, &
-                                              ProvideArray2dDouble,&
-                                              ProvideFieldDouble
+        generic, public :: ProvideVariable => ProvideFieldDouble
```

Comment:
Please could you update the comment above to explain why only one routine is now listed here.

Reply:
Done.